src/main/fc/mw.c

```diff
@@ -829,25 +819,12 @@ void taskMainPidLoopCheck(void)
 
     const uint32_t startTime = micros();
     while (true) {
-        if (gyroSyncCheckUpdate(&gyro) || ((currentDeltaTime + (micros() - previousTime)) >= (gyro.targetLooptime + GYRO_WATCHDOG_DELAY))) {
-            static uint8_t pidUpdateCountdown;
-
+        if (isGyroDataReady() || ((currentDeltaTime + (micros() - previousTime)) >= (targetPidLooptime + GYRO_WATCHDOG_DELAY))) {
```

Comment:
This is so much nicer. I'd change it to the equivalent:

```
const uint32_t startTime = micros();
while (true) {
    if (isGyroDataReady() || ((currentDeltaTime + (micros() - previousTime)) >= (targetPidLooptime + GYRO_WATCHDOG_DELAY))) {
        if (debugMode == DEBUG_PIDLOOP) {debug[0] = micros() - startTime;} // time spent busy waiting
        break;
    }
}
subTaskMainSubprocesses();
subTaskPidController();
subTaskMotorUpdate();
```

which makes the busy loop obvious.